### ISSUE TITLE
How to represent a read-only Key-Value store?

### ISSUE THREAD

**Issue by mpwarres** [WebAssembly/wasi-keyvalue#43]
What would the recommended way to represent a read-only key-value store (populated from or fronting some other source) with these interfaces--have it export `store` but return errors from `bucket.set` and `.delete` calls? Is it worth considering having a read-only `store` interface, or is this use case too niche to warrant a separate interface?

**Comment by Mossaka** [comment]
What will be the use cases for read-only key-value stores? Will [`wasi:runtime-config`](https://github.com/WebAssembly/wasi-runtime-config/tree/main) solve these use cases, which is a simply interface that provides read-only operations.

**Comment by thomastaylor312** [comment]
Yeah I'm definitely curious on the use case here. With the current interface, since it returns errors, you could create an implementation that returns an error on write ops. But I also understand you may want to have the interface indicate the difference
